connectors/services/sync.py

```diff
@@ -64,32 +71,61 @@ async def _one_sync(self, connector, es, sync_now):
 
         try:
             # the heartbeat is always triggered
-            connector.start_heartbeat(self.hb)
+            await connector.heartbeat(self.heartbeat_interval)
 
             logger.debug(f"Connector status is {connector.status}")
 
             # we trigger a sync
             if connector.status in (Status.CREATED, Status.NEEDS_CONFIGURATION):
                 # we can't sync in that state
                 logger.info(f"Can't sync with status `{e2str(connector.status)}`")
+                return
+
+            if connector.features.sync_rules_enabled():
+                await validate_filtering(
+                    connector, self.connectorIndex, ValidationTarget.DRAFT
+                )
+
+            if not sync_now:
+                next_sync = connector.next_sync()
+                if next_sync == SYNC_DISABLED or next_sync - self.idling > 0:
```

Comment:
2 things here: 

* Why do need the second condition against `SYNC_DISABLED`?
* I would move this logic (the entire block from L89 to L100  into a separate method

Reply:
I think the inner if-else branch is to log different messages. I just copied this block from somewhere else. 

Will furthur refactor it.

Reply:
It can be a big refactoring overall (given we want to test it fully, I think there are missing tests for scheduling), so can probably be done in a separate PR